cli/cmd/create-workspace.go

```diff
@@ -0,0 +1,149 @@
+			Args:  cobra.RangeArgs(1, 1),
+			Long: out.Long(`Create a workspace in Codesphere.
+
+				Specify a (private) git repository or start an empty workspace.
+				Environment variables can be set to initialize the workspace with a specific environment.
+				The command will wait for the workspace to become running or a timeout is reached.
+
+				To decide which plan suits your needs, run 'cs list plans'
+			`),
+			Example: out.FormatExampleCommands("create workspace my-workspace", []out.Example{
+				{Cmd: "-p 20", Desc: "Create an empty workspace, using plan 20"},
+				{Cmd: "-r https://github.com/codesphere-cloud/landingpage-temp.git", Desc: "Create a workspace from a git repository"},
+				{Cmd: "-r https://github.com/codesphere-cloud/landingpage-temp.git -e FOO=BAR -e A=B", Desc: "Create a workspace and set environment variables"},
+				{Cmd: "-r https://github.com/codesphere-cloud/landingpage-temp.git --vpn myVpn", Desc: "Create a workspace and connect to VPN myVpn"},
+				{Cmd: "-r https://github.com/codesphere-cloud/landingpage-temp.git --timeout 30", Desc: "Create a workspace and wait 30 seconds for it to become running"},
+				{Cmd: "-r https://github.com/codesphere-cloud/landingpage-temp.git -b staging", Desc: "Create a workspace from branch 'staging'"},
+				{Cmd: "-r https://github.com/my-org/my-private-project.git -P", Desc: "Create a workspace from a private git repository"},
+			}),
+		},
+		Opts: CreateWorkspaceOpts{GlobalOptions: opts},
+	}
+	workspace.Opts.Repo = workspace.cmd.Flags().StringP("repository", "r", "", "Git repository to create the workspace from")
+	workspace.Opts.Vpn = workspace.cmd.Flags().String("vpn", "", "Vpn config to use")
+	workspace.Opts.Env = workspace.cmd.Flags().StringArrayP("env", "e", []string{}, "Environment variables to set in the workspace")
+	workspace.Opts.Plan = workspace.cmd.Flags().IntP("plan", "p", 8, "Plan ID for the workspace")
+	workspace.Opts.Private = workspace.cmd.Flags().BoolP("private", "P", false, "Use private repository")
+	workspace.Opts.Timeout = workspace.cmd.Flags().Duration("timeout", 60*time.Second, "Time to wait for the workspace to start")
+	workspace.Opts.Branch = workspace.cmd.Flags().StringP("branch", "b", "", "branch to check out")
+
+	create.AddCommand(workspace.cmd)
+	workspace.cmd.RunE = workspace.RunE
+}
+
+func (c *CreateWorkspaceCmd) CreateWorkspace(client Client, teamId int, wsName string) (*api.Workspace, error) {
+	envVars, err := cs.ArgToEnvVarMap(*c.Opts.Env)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse environment variables: %w", err)
+	}
+
+	args := api.DeployWorkspaceArgs{
+		TeamId:  teamId,
+		PlanId:  *c.Opts.Plan,
+		Name:    wsName,
+		EnvVars: envVars,
+
+		IsPrivateRepo: *c.Opts.Private,
+
+		Timeout: *c.Opts.Timeout,
+	}
+
+	if c.Opts.Repo != nil && *c.Opts.Repo != "" {
+		validatedUrl, err := cs.ValidateUrl(*c.Opts.Repo)
+		if err != nil {
+			return nil, fmt.Errorf("validation of repository URL failed: %w", err)
```

Comment:
a bit dupliocated error message, but fine for now

Reply:
this one is more specific about being a repo URL that failed, the other is generic URL validation.